src/openforms/authentication/migrations/0002_migrate_submission_data.py

```diff
@@ -0,0 +1,76 @@
+# Generated by Django 3.2.15 on 2022-08-12 12:15
+
+from django.db import migrations
+from django.db.models import Q
+
+from ..constants import AuthAttribute
+
+
+def forward(apps, schema_editor):
+    AuthInfo = apps.get_model("of_authentication", "AuthInfo")
+    Submission = apps.get_model("submissions", "Submission")
+
+    submissions = Submission.objects.filter(~Q(auth_plugin=""))
+
+    auth_info_to_create = []
+    for submission in submissions:
+        if hasattr(submission, "auth_info"):
+            continue
+
+        value = submission.bsn or submission.kvk or submission.pseudo
```

Comment:
that'll always pick `submission.bsn` if it's hashed already. I think you need to select the appropriate attribute based on the plugin used to grab the correct field.

Probably best to hardcode in the migration which plugin ID maps to which `AuthAttribute`